### PR TITLE
issue #270 #OT-171

### PR DESCRIPTION
**Pull request by polishcode**
tiny refactoring of android docker file

@justusranvier this is sth I've done while patching core issue; basically extracted parameters, removed unnecessary android SDK component

### CODE REVIEW

File: android/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,13 @@
+#single point for ARG definitions
+ARG ANDROID_TOOLS=8092744
+ARG ANDROID_LEVEL=31
+ARG ANDROID_LEVEL_TOOLCHAIN=31
+ARG ANDROID_BUILD_TOOLS=31.0.0
+ARG NDK=23.1.7779620
+ARG JOBS=4
```
Comment (justusranvier) [inline android/Dockerfile]: The default JOBS argument should match the environment where we plan to automatically build the images. Previously this was Dockerhub although that was never implemented.

File: android/Dockerfile [hunk 2]
```diff
@@ -22,20 +32,25 @@ RUN dnf install -y \
...
+    cd /opt/android && \
+    wget "https://dl.google.com/android/repository/commandlinetools-linux-${ANDROID_TOOLS}_latest.zip" && \
+    unzip "commandlinetools-linux-${ANDROID_TOOLS}_latest.zip" && \
+    rm "commandlinetools-linux-${ANDROID_TOOLS}_latest.zip" && \
+    yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux --update && \
+    yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "ndk;${NDK}" && \
+    yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "platforms;android-${ANDROID_LEVEL}" && \
+    yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "build-tools;${ANDROID_BUILD_TOOLS}" && \
+    find /opt/android-sdk-linux/ndk -name libz.a -delete
+#optimize image, the rest is not needed for now
```
Comment (justusranvier) [inline android/Dockerfile:51]: Can these lines be removed entirely if they are not needed?

File: android/Dockerfile (hunk 1 above)
Comment (polishcode) [reply android/Dockerfile]: Previously hardcoded value was 2, increased it a bit for my local machine. Generally the whole script is far from being done. I am not a docker expert (1 day of experience :) ) but it seems like a lot of e.g. caching could be introduced (e.g. qt builds). I would leave it like that.

File: android/Dockerfile (hunk 2 above)
Comment (polishcode) [reply android/Dockerfile:51]: I left those as reference for the time we start working on running android tests on qemu.
Please treat this PR as an iterative improvement.

File: android/Dockerfile (hunk 1 above)
Comment (justusranvier) [reply android/Dockerfile]: You can increase it locally by passing --build-arg "JOB=X" on the command line without modifying the dockerfile itself

Comment (polishcode) [reply android/Dockerfile]: Ok, did not investigate that. I will restore its value to 2.

**Pull request merged**